### PR TITLE
Enabling new metrics

### PR DESCRIPTION
**Pull request by busecolak**
This PR:
* Enabled realtime metrics emitted from MiddleManager.
* Added taskID label to realtime metrics.
* Added new counter metrics emitted from Coordinator.
* Added new histogram metrics emitted from Coordinator and MiddleManager.
* Added a sample Dockerfile
* Is tested with Druid Cluster with Kafka Indexing Service enabled.

### CODE REVIEW

**Comment by elukey** [comment]
@busecolak this is really great, I'll try to review this code asap. What version of Druid are you using? We are still at 0.12.3 (super old I know) so I will probably not be able to test it with a more recent version. I'll probably try something with Docker in case, thanks a lot for the Dockerfile!

**Comment by busecolak** [comment]
@elukey you're welcome. We are using Druid 0.17.0 version, as a dockerized cluster.

**Comment by elukey** [comment]
@busecolak really sorry for the delay, busy days!

I reviewed your patch and it looks really good, but then I started thinking how to support multiple versions of Druid without every time making such invasive changes. I started implementing some ideas and I ended up with this prototype:

https://gerrit.wikimedia.org/r/#/c/operations/software/druid_exporter/+/592261/

(this is a read-only mirror of Gerrit, any patch needs to go there to be merged)

Basically the idea is to abstract away all the logic to handle metrics and let the user configure it via JSON. I haven't tested the code yet, but if you could check and tell me your thoughts I'd be really happy. My hope is that after this change any Druid version would be supported, simply changing the number/type of metrics in the config file (rather than in the code).

**Comment by busecolak** [comment]
@elukey that's really good idea. And the prototype looks good. It can support any Druid version. It's nice to be able to modify metrics from the config file. 

I have reviewed and tested your patch (only using config file you provided), I have added some comments. Except some minor problems, it seems to be working. However, I have concerns regarding performance especially on production usage. Maybe you can consider to use multithreading when collecting and updating the metrics.

**Comment by elukey** [comment]
@busecolak merged the refactoring, if you still have time/patience I'd love to get a pull request for the Docker image and the new metrics :)

**Comment by busecolak** [comment]
@elukey hello again, many thanks for the refactoring. I have updated the PR to include followings:

- Applied refactoring changes 
- Example config file for metrics emitted from Druid version 0.17.0
- Dockerfile including example config
- Updated ReadMe with Docker commands

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.8-alpine
```
Comment (elukey) [inline Dockerfile:1]: I would, if possible, allow people to configure the image with something like `FROM ${image_name}`

File: druid_exporter/collector.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
 class DruidCollector(object):
     scrape_duration = Summary(
-            'druid_scrape_duration_seconds', 'Druid scrape duration')
```
Comment (elukey) [inline druid_exporter/collector.py:28]: Leftover?

**Comment by elukey** [comment]
@busecolak I added a couple of comments, looks good! There are two options:
* I copy your diff in a gerrit code review under my user, adding credits in the commit msg
* you submit a code patch in gerrit and after review I can merge them. It should be a simple as cloning the repo, applying the diff and then doing `git push origin HEAD:refs/for/master` to create the code review.

Let me know!

**Comment by busecolak** [comment]
Hello @elukey, thanks for the review!

I submitted a patch (fixed your comments above) and created the code review in gerrit.
https://gerrit.wikimedia.org/r/#/c/operations/software/druid_exporter/+/594408/

**Comment by elukey** [comment]
@busecolak merged thanks a lot for all the work!

**Comment by busecolak** [comment]
@elukey great! 
Since the refactoring changes and example config files are merged, i am closing this PR.
